### PR TITLE
ci: do_not_merge: convert to python, pass instead of skip

### PR DESCRIPTION
**Pull request by fabiobaltieri**
I've been troubleshooting some PR not showing on the merge list dashboard and I have a vague feeling that one of the issue could be GitHub mishandling workflows that fails and then skip but never succeed.

Try to address this by converting the do not merge run to something that passes instead of skipping, and also rewrite it in python while at it, which is probably unnecessary but kindof nice if we want to add more complex conditions down the road.

### CODE REVIEW

**Comment by fabiobaltieri** [comment]
Not too sure it'll actually help but I think it's nice to have this in python anyway

File: .github/workflows/do_not_merge.yml [hunk 1]
```diff
@@ -6,15 +6,22 @@ on:
...
+
+      - name: Set up Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: 3.11
+
+      - name: Install Python dependencies
+        run: |
+          sudo pip3 install -U setuptools wheel pip
+          pip3 install -U PyGithub
```
Comment (pdgendt) [inline .github/workflows/do_not_merge.yml:23]: Why `sudo` and why the other packages?
The action docs specifies simply running `pip`

```suggestion
        run: |
          pip install PyGithub
```

Comment (fabiobaltieri) [reply .github/workflows/do_not_merge.yml]: ok fixed, let's sync the other workflows separately

**Review by pdgendt — APPROVED** [review]

File: .github/workflows/do_not_merge.yml [hunk 2]
```diff
@@ -6,15 +6,21 @@ on:
...
+      - name: Install Python dependencies
+        run: |
+          pip install PyGithub
+
+      - name: Run the check script
         run: |
-          echo "Pull request is labeled as 'DNM', 'TSC', 'Architecture Review' or 'dev-review'."
-          echo "This workflow fails so that the pull request cannot be merged."
-          exit 1
+          ./scripts/ci/do_not_merge.py -p "${{ github.event.pull_request.number }}"
```
Comment (stephanosio) [inline .github/workflows/do_not_merge.yml:26]: Not really sure why we need to convert this simple workflow into a Python script. If we ever need to handle more complex conditions, that can be done within the workflow itself using the [`github-script` action](https://github.com/actions/github-script/).

File: scripts/ci/do_not_merge.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    return parser.parse_args(argv)
+
+
+def main(argv):
+    args = parse_args(argv)
+
+    token = os.environ.get('GITHUB_TOKEN', None)
+    gh = github.Github(token)
+    repo = gh.get_repo("zephyrproject-rtos/zephyr")
+    pr = repo.get_pull(args.pr)
```
Comment (stephanosio) [inline scripts/ci/do_not_merge.py:33]: Let us try not to waste the already scarce API call quota with these unnecessary calls ...

Comment (fabiobaltieri) [reply scripts/ci/do_not_merge.py:33]: Sure, I'll close this redo it just making it pass instead of skip.

**Pull request closed without merging**